python/cudf/cudf/_lib/csv.pyx

```diff
@@ -425,10 +431,19 @@ cpdef write_csv(
     cdef unique_ptr[data_sink] data_sink_c
     cdef sink_info sink_info_c = make_sink_info(path_or_buf, data_sink_c)
 
-    if header is True and table._column_names is not None:
-        metadata_.column_names.reserve(len(table._column_names))
-        for col_name in table._column_names:
-            metadata_.column_names.push_back(str(col_name).encode())
+    if header is True and len(all_names) > 0:
+        metadata_.column_names.reserve(len(all_names))
+        if len(all_names) == 1:
+            if all_names[0] in (None, ''):
```

Comment:
This behavior is a bit wonky from Pandas. This is to handle when we're only writing the index and it's not a MultiIndex, right?

Reply:
Not actually, it is a generic case. So what pandas will do is, if we are writing a single column to a csv file, which has no column name, the header will have a double quote(`""`). But when there are multiple columns to be written and any of their names is either `None` or `''`, only an empty string(`''`) gets written.

So the cases when we would run in single column being written to a csv file are:
1. Dataframe has one column and `index` is False.
2. Dataframe has no columns but has a valid index and `index` is True.